### PR TITLE
[INLONG-8663][Manager] Add tenant related OpenAPI

### LINKED ISSUE #8663: [Improve][Manager] Add tenant related OpenAPI
### Description

Add all tenant related openAPI

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by castorqin**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8663  

### Motivation

Manager component Add tenant related openAPI

### Modifications
The details can be found at [#8663  ](https://github.com/apache/inlong/issues/8663).

### CODE REVIEW

**Review by healchow — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInLongClusterController.java [hunk 1]
```diff
@@ -80,6 +87,8 @@ public Response<ClusterTagResponse> getTag(@PathVariable Integer id) {
     public Response<List<ClusterTagResponse>> listTag(@RequestBody ClusterTagPageRequest request) {
         Preconditions.expectNotNull(request, ErrorCodeEnum.INVALID_PARAMETER, "request cannot be null");
         Preconditions.expectNotNull(LoginUserUtils.getLoginUser(), ErrorCodeEnum.LOGIN_USER_EMPTY);
+        request.setIsAdminRole(
+                LoginUserUtils.getLoginUser().getRoles().contains(TenantUserTypeEnum.TENANT_ADMIN.name()));
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInLongClusterController.java:91]: Both INLONG_ADMIN and TENANT_ADMIN are admin role for this interface

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInLongClusterController.java [hunk 2]
```diff
@@ -125,6 +134,8 @@ public Response<ClusterInfo> get(@PathVariable Integer id) {
     public Response<List<ClusterInfo>> list(@RequestBody ClusterPageRequest request) {
         Preconditions.expectNotNull(request, ErrorCodeEnum.INVALID_PARAMETER, "request cannot be null");
         Preconditions.expectNotNull(LoginUserUtils.getLoginUser(), ErrorCodeEnum.LOGIN_USER_EMPTY);
+        request.setIsAdminRole(
+                LoginUserUtils.getLoginUser().getRoles().contains(TenantUserTypeEnum.TENANT_ADMIN.name()));
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInLongClusterController.java:138]: ditto

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantRoleController.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    @RequestMapping(value = "/role/tenant/list", method = RequestMethod.POST)
+    @ApiOperation(value = "List tenant roles by paginating")
+    public Response<PageResult<TenantRoleInfo>> listByCondition(@RequestBody TenantRolePageRequest request) {
+        return Response.success(tenantRoleService.listByCondition(request));
+    }
+
+    @RequestMapping(value = "/role/tenant/delete/{id}", method = RequestMethod.DELETE)
+    @ApiOperation(value = "Delete tenant role by ID")
+    @ApiImplicitParam(name = "id", dataTypeClass = Integer.class, required = true)
+    @RequiresRoles(logical = Logical.OR, value = {UserRoleCode.TENANT_ADMIN, UserRoleCode.INLONG_ADMIN})
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantRoleController.java:87]: Only INLONG ADMIN can delete tenants

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantRoleController.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    public Response<Integer> save(@Validated @RequestBody TenantRoleRequest request) {
+        String operator = LoginUserUtils.getLoginUser().getName();
+        return Response.success(tenantRoleService.save(request, operator));
+    }
+
+    @RequestMapping(value = "/role/tenant/update", method = RequestMethod.POST)
+    @OperationLog(operation = OperationType.CREATE)
+    @ApiOperation(value = "Update tenant role")
+    public Response<Boolean> update(@Validated @RequestBody TenantRoleRequest request) {
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantRoleController.java:73]: Only INLONG_ADMIN or TENANT_ADMIN can update tenant info

**Review by vernedeng — CHANGES_REQUESTED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantController.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    @ApiOperation(value = "Update inlong tenant")
+    public Response<Boolean> update(@Validated(UpdateByIdValidation.class) @RequestBody InlongTenantRequest request) {
+        return Response.success(tenantService.update(request));
+    }
+
+    @RequestMapping(value = "/tenant/delete/{name}", method = RequestMethod.GET)
+    @ApiOperation(value = "Delete inlong tenant by name")
+    @ApiImplicitParam(name = "name", dataTypeClass = String.class, required = true)
+    public Response<Boolean> delete(@PathVariable String name) {
+        return Response.success(tenantService.delete(name));
```
Comment (luchunliang) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantController.java:81]: Check whether the tenant has been refered by business before deleting.

Comment (castorqin) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantController.java:81]: create a new issue to fix it

**Review by vernedeng — APPROVED** [review]

**Pull request merged**